File: cs/libldt3/model/objekte/Diagnose.cs
/*
 * Copyright 2016-2024  Christoph Brill <opensource@christophbrill.de>
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
using libldt3.attributes;
using libldt3.model;
using libldt3.model.enums;
using libldt3.model.regel.format;

namespace libldt3
{
    namespace model
    {
        namespace objekte
        {
            /// <summary>
            /// Mit diesem Objekt können Angaben zu Diagnosen des Patienten übertragen werden.
            /// </summary>
            [Objekt(Value = "0100")]
            public class Diagnose : Kontext
            {
                [Objekt]
                public class IcdCode_ : Kontext
                {
                    public string Value;
                    [Feld(Value = "6003", Feldart = Feldart.bedingt_kann)]
                    [Regelsatz(Laenge = 1)]
                    public Diagnosesicherheit? Diagnosesicherheit;
                    [Feld(Value = "6004", Feldart = Feldart.bedingt_kann)]
                    [Regelsatz(Laenge = 1)]
                    public Lokalisation? Lokalisation;
                    [Feld(Value = "6006", Feldart = Feldart.bedingt_kann)]
                    [Regelsatz(MaxLaenge = 60)]
                    public IList<string> Diagnoseerlaeuterung;
                    [Feld(Value = "6008", Feldart = Feldart.bedingt_kann)]
                    [Regelsatz(MaxLaenge = 60)]
                    public IList<string> Diagnoseausnahmetatbestand;

                }
                [Feld(Value = "4207", Feldart = Feldart.kann)]
                [Regelsatz(MaxLaenge = 60)]
                public IList<string> DiagnoseVerdachtsdiagnose;
                [Feld(Value = "6001", Feldart = Feldart.kann)]
                [Regelsatz(Value = new[] { typeof(F004) }, MinLaenge = 3, MaxLaenge = 6)]
                public IcdCode_ IcdCode;

            }
        }
    }
}

File: java/src/main/java/libldt3/model/objekte/Diagnose.java
/*
 * Copyright 2016-2024  Christoph Brill &lt;opensource@christophbrill.de&gt;
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package libldt3.model.objekte;

import java.util.List;

import libldt3.annotations.Feld;
import libldt3.annotations.Feldart;
import libldt3.annotations.Objekt;
import libldt3.annotations.Regelsatz;
import libldt3.model.Kontext;
import libldt3.model.enums.Diagnosesicherheit;
import libldt3.model.enums.Lokalisation;
import libldt3.model.regel.format.F004;

/**
 * Mit diesem Objekt können Angaben zu Diagnosen des Patienten übertragen werden.
 */
@Objekt("0100")
public class Diagnose implements Kontext {

    @Objekt
    public static class IcdCode implements Kontext {
        @SuppressWarnings("unused")
        public String value;
        @Feld(value = "6003", feldart = Feldart.bedingt_kann)
        @Regelsatz(laenge = 1)
        public Diagnosesicherheit diagnosesicherheit;
        @Feld(value = "6004", feldart = Feldart.bedingt_kann)
        @Regelsatz(laenge = 1)
        public Lokalisation lokalisation;
        @Feld(value = "6006", feldart = Feldart.bedingt_kann)
        @Regelsatz(maxLaenge = 60)
        public List<String> diagnoseerlaeuterung;
        @Feld(value = "6008", feldart = Feldart.bedingt_kann)
        @Regelsatz(maxLaenge = 60)
        public List<String> diagnoseausnahmetatbestand;
    }

    @Feld(value = "4207", feldart = Feldart.kann)
    @Regelsatz(maxLaenge = 60)
    public List<String> diagnoseVerdachtsdiagnose;
    @Feld(value = "6001", feldart = Feldart.kann)
    @Regelsatz(value = F004.class, minLaenge = 3, maxLaenge = 6)
    public IcdCode icdCode;

}
